frontend/src/user/profiles/UserProfile.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import { Nav, Container, Loader, Notification } from '@components/';
+import { UserIcon } from '@icons/';
+import { RootState } from '@store/store';
+import { FollowerRecord, FollowingRecord } from '@store/user';
+import {
+  followUser,
+  getUser,
+  getUserFollowers,
+  getUserFollowing,
+  unfollowUser,
+} from '@api/UserApi';
+import UserProfileItem from './UserProfileItem';
+import { setIsError, setIsLoading } from '@store/category';
+
+interface SpecificUser {
+  id?: number;
+  name?: string;
+  user_image?: string;
+  email?: string;
+  followers?: FollowerRecord[];
+  following?: FollowingRecord[];
+}
+
+export default function UserProfile() {
+  const { user_id } = useParams();
+  const [cookies, _] = useCookies();
+  const dispatch = useDispatch();
+  const state = useSelector((state: RootState) => state);
+  const [user, setUser] = useState<SpecificUser>();
+  const [isViewingFollowers, setIsViewingFollowers] = useState(true);
+  const isFollowing = JSON.parse(
+    localStorage.getItem('user_following') || '{}'
```

Comment:
Confirmation:
* Why are we manipulating the localStorage?

Reply:
i'm storing the retrieved user_following (users that currently logged in user is following) because when I tried to store it in the state, the values are gone when I refresh the page. the user_following value is used to indicate if the followers/following of another user is also being followed by the current user
![image](https://user-images.githubusercontent.com/86587091/151129267-0e1bb028-1f08-4960-bf78-e67bb390b1c4.png)